### PR TITLE
(Re)load FITS in the same window

### PR DESCRIPTION
**Pull request by hombit**
This for the first part of #8

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -74,9 +74,15 @@ if(APPLE)
...
+		# This is true for Makefile generators only
+		set(BUNDLE_DIR ${CMAKE_CURRENT_BINARY_DIR}/Fips.app)
+	else()
+		set(BUNDLE_DIR $<TARGET_BUNDLE_DIR:${TARGET}>)
+	endif(${CMAKE_VERSION} VERSION_LESS 3.9.0)
 	add_custom_command(TARGET ${TARGET}
 		POST_BUILD
-		COMMAND macdeployqt ARGS "${CMAKE_CURRENT_BINARY_DIR}/Fips.app" -always-overwrite
+		COMMAND macdeployqt ARGS ${BUNDLE_DIR} -always-overwrite
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
```
Comment (matwey) [inline CMakeLists.txt]: Where did it came from?
Why do we need it for this particular PR?
Why is it not in a separate commit?

File: include/exception.h [hunk 2]
```diff
@@ -19,6 +19,7 @@
 #ifndef _EXCEPTION_H_
 #define _EXCEPTION_H_
 
+#include <QtGui/qopengl.h>
```
Comment (matwey) [inline include/exception.h]: I suppose it should be in other file.

File: include/mainwindow.h [hunk 3]
```diff
@@ -63,7 +63,11 @@ class MainWindow:
 	static constexpr double zoomOut_factor_ = 0.8;
 	static constexpr const char homepage_url_[] = "http://fips.space";
 
+	QString fits_filename_;
 	std::unique_ptr<FITS> fits_;
+	std::unique_ptr<FITS> loadFITS(const QString& fits_filename) const;
+	void openInThisWindow(const QString& fits_filename);
+	void setWindowTitle();
```
Comment (matwey) [inline include/mainwindow.h]: Isn't it better to call it `updateWindowTitle`?

File: include/scrollzoomarea.h [hunk 4]
```diff
@@ -20,6 +20,7 @@
 #define _SCROLLZOOMAREA_H
 
 #include <QAbstractScrollArea>
+#include <QMessageBox>
```
Comment (matwey) [inline include/scrollzoomarea.h]: Move include to the module where it is actually required

File: src/openglerrors.cpp [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+}
+QException* OpenGLException::clone() const {
+	return new OpenGLException(*this);
+}
+// From OpenGL ES 2.0 documentation:
+// https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glGetError.xml
+QString OpenGLException::glErrorString(GLenum gl_error_code) {
+	switch (gl_error_code) {
+//		case GL_NO_ERROR:
+//			return "No error has been recorded. The value of this symbolic constant is guaranteed to be 0.";
```
Comment (matwey) [inline src/openglerrors.cpp]: Do we really want to show "Unknown error" instead of "No error"?

File: CMakeLists.txt (hunk 1 above)
Comment (hombit) [reply CMakeLists.txt]: - I'd like to run the project in Xcode and used Xcode CMake generator. It putted macOS bundle into another directory then Makefile CMake generator. CMake 3.9.0+ provides possibility to get macOS bundle directory.
- I used Xcode for profiling. Profiling helped us to make a decision to decline implementation with OpenGLWidget reconstruction and make new implementation with texture and shader reloading.
- I'll make separate commit with this changes.

File: include/exception.h (hunk 2 above)
Comment (hombit) [reply include/exception.h]: ok

File: include/mainwindow.h (hunk 3 above)
Comment (hombit) [reply include/mainwindow.h]: ok

File: include/scrollzoomarea.h (hunk 4 above)
Comment (hombit) [reply include/scrollzoomarea.h]: ok

File: src/openglerrors.cpp (hunk 5 above)
Comment (hombit) [reply src/openglerrors.cpp]: The short answer is yes.
Some changes in 14d09522a126c056a13d882abe9985215adb9b20 in shader program initialisation procedure are wrong, e.g. look [this line](https://github.com/matwey/fips3/pull/52/commits/14d09522a126c056a13d882abe9985215adb9b20#diff-0545048cc4207fb82d23cb02af3a4669L213). It looks like Qt checks for OpenGL error itself, prints description of the error for user and returns us error status as bool. glGetError() returns GL_NO_ERROR in such cases and we should show "Unknown error". 
I will revert these changes back.

**Pull request merged**